packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterForm.tsx

```diff
@@ -297,16 +312,38 @@ const GridFilterForm = React.forwardRef<HTMLDivElement, GridFilterFormProps>(
           column.filterOperators![0];
 
         // Erase filter value if the input component or filtered column type is modified
-        const eraseItemValue =
+        const eraseFilterValue =
           !newOperator.InputComponent ||
           newOperator.InputComponent !== currentOperator?.InputComponent ||
           column.type !== currentColumn!.type;
 
+        let filterValue = eraseFilterValue ? undefined : item.value;
+
+        // Check filter value against the new valueOptions
+        if (column.type === 'singleSelect' && filterValue !== undefined) {
```

Comment:
Similar to the one above - we need to remove values that do not exist in the new valueOptions.